### DISCUSSION TITLE
Philosophy on brew vs. pip for py3 pkgs?

### DISCUSSION THREAD

**Discussion by edrozenberg** [Homebrew/discussions#3593] [Everyday usage] [answered]
I've searched old discussions and elsewhere to see what the HB general approach/thinking is regarding whether to install python3 packages using `brew` or `pip3`

For example Ansible is available both via `brew` and via `pip3`. As is iPython and many other things.

Some aspects I'm aware of:
- brew installs py pkgs in Cellar not site-packages, so they continue to work even when changing major py versions (3.9 -> 3.10).
- pip3 installs in `.../lib/python<version>/site-packages`, so pkgs need to be reinstalled when py major ver changes.
- pip3 is more likely to pull the newest version of packages sooner than brew, since pip pulls from the canonical pypy repos.
- If "mixing", installing some pkgs with brew and some with pip, it can get a bit confusing to track what python pkgs are installed, since some will be in Cellar, and some in site-packages. Sometimes it's nice to have a list of all installed py packages in a single place, esp. when making a list of what to install on other machines.

There's an older unanswered discussion here that was asking similar/related questions:
https://github.com/Homebrew/discussions/discussions/293

**Comment by SMillerDev** [comment, marked as answer]
https://docs.brew.sh/Formula-Cookbook#specifying-gems-python-modules-go-projects-etc-as-dependencies seems to be the answer

**Reply by edrozenberg** [reply]
> Homebrew doesn’t package already-packaged language-specific libraries. These should be installed directly from gem/cpan/pip etc.

@SMillerDev that answers it but brew still offers python stuff:
ansible, numpy, scipy, ipython, and others

Based on that forum post I'll go ahead and remove any brewed python stuff and use pip3 instead. Thanks!
